examples/buffer_test.rs

```diff
@@ -0,0 +1,351 @@
+use std::{error, fmt, fs::File, os::unix::io::OwnedFd, path::PathBuf};
+
+use clap::{Parser, ValueEnum};
+
+use smithay::{
+    backend::{
+        allocator::{
+            dmabuf::{AsDmabuf, Dmabuf},
+            gbm::GbmDevice,
+            vulkan::{ImageUsageFlags, VulkanAllocator},
+            Allocator, Fourcc, Modifier,
+        },
+        drm::{DrmDevice, DrmDeviceFd, DrmNode},
+        egl::{EGLContext, EGLDevice, EGLDisplay},
+        renderer::{gles2::Gles2Renderer, Bind, Frame, ImportDma, Renderer},
+        vulkan::{version::Version, Instance, PhysicalDevice},
+    },
+    utils::{DeviceFd, Rectangle, Transform},
+};
+
+#[derive(Parser, Debug)]
+#[command(author, version, about, long_about = None, disable_help_flag = true)]
+struct Args {
+    #[arg(short, long)]
+    allocator: AllocatorType,
+
+    #[arg(short, long)]
+    import: String,
+    #[arg(short, long)]
+    export: Option<String>,
+    #[arg(short, long)]
+    renderer: Option<RendererType>,
+
+    #[arg(short, long, default_value_t = 256)]
+    width: u32,
+    #[arg(short, long, default_value_t = 256)]
+    height: u32,
+    #[arg(short, long, default_value = "Abgr8888")]
+    fourcc: String,
```

Comment:
I believe clap allows specifying an attribute to use some function to do the `fn(String) -> T` conversion so you can store a DrmFourcc code directly here.

Reply:
Done.